src/commands.cpp

```diff
@@ -1934,38 +1938,64 @@ void CommandGetUA::procresult()
         error e = (error)client->json.getint();
 
 #ifdef USE_SODIUM
-        if ((e == API_ENOENT) && (user->userhandle == client->me)
-                && ((priv && strncmp(attributename.c_str(), "prEd255", 7))
-                        || (!priv && strncmp(attributename.c_str(), "puEd255", 7))))
+        if ((e == API_ENOENT) && (user->userhandle == client->me))
         {
-            // We apparently don't have Ed25519 keys, yet. Let's make 'em.
-            if(!client->inited25519())
-            {
-                return(client->app->getua_result(API_EINTERNAL));
-            }
 
-            // Return the required key data.
-            if (strncmp(attributename.c_str(), "prEd255", 7))
+            if (attributename == "*prEd255" || attributename == "+puEd255")
```

Comment:
Now I know where the naming of `prEd255` comes from. That was way early design, before we decided together with Mathias to store private attributes in containers/collections, as the crypto overhead is too big for many little private attributes. Sorry for that, the whole libsodium stuff is pretty outdated.

Michael Holmwood was working on that, but unfortunately it never got fully finished/merged :-(